compute/cloud-client/src/main/java/compute/Quickstart.java

```diff
@@ -0,0 +1,216 @@
+import com.google.cloud.compute.v1.InstancesScopedList;
+import com.google.cloud.compute.v1.NetworkInterface;
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.Operation.Status;
+import com.google.cloud.compute.v1.ZoneOperationsClient;
+import java.io.IOException;
+import java.util.Map.Entry;
+
+public class Quickstart {
+
+  // [START compute_instances_create]
+
+  /**
+   * Creates a GCP Compute Instance
+   *
+   * @param project     Google Cloud Project id
+   * @param zone        Google Cloud Project zone eg: "us-central1-a"
+   * @param machineType Type of the instance that you want to create, eg: "n1-standard-1"
+   * @param machineName Name to identify the machine/ instance
+   * @param sourceImage Image to be mounted in the machine. eg: "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-7-wheezy-v20150710"
+   * @param diskSizeGb  Size of the disk to be allocated for the machine. eg: "10" in GBs
+   */
+
+  public void createInstance(String project, String zone, String machineName, String machineType,
+      String sourceImage, String diskSizeGb)
+      throws IOException {
+
+    try (InstancesClient instancesClient = InstancesClient.create()) {
+      // Instance creation requires at least one persistent disk and one network interface
+      AttachedDisk disk =
+          AttachedDisk.newBuilder()
+              .setBoot(true)
+              .setAutoDelete(true)
+              .setType(Type.PERSISTENT)  // Can be set to either 'SCRATCH' or 'PERSISTENT'
+              .setInitializeParams(
+                  AttachedDiskInitializeParams.newBuilder().setSourceImage(sourceImage)
+                      .setDiskSizeGb(diskSizeGb).build())
+              .build();
+
+      // "default" network interface is created automatically for every project
+      NetworkInterface networkInterface = NetworkInterface.newBuilder().setName("default").build();
+
+      // Bind machine_name, machine_type, disk and network interface to an instance
+      Instance instanceResource =
+          Instance.newBuilder()
+              .setName(machineName)
+              .setMachineType(machineType)
+              .addDisks(disk)
+              .addNetworkInterfaces(networkInterface)
+              .build();
+
+      // Inserting the instance in the specified project and zone
+      Operation response = instancesClient.insert(project, zone, instanceResource);
+
+      // Waits until the insertion operation is completed
+      while (response.getStatus() != Status.DONE) {
+        ZoneOperationsClient zoneOperationsClient = ZoneOperationsClient.create();
+        response = zoneOperationsClient.wait(project, zone, response.getId());
+      }
+
+      if (response.getError().toString() == "") {
+        System.out.println("Instance creation successful ! ! ");
+      } else {
+        System.out.println("Instance creation failed ! ! " + response.getError());
+      }
```

Comment:
It's preferable to look for the negative case and return early to avoid branching:

```suggestion
      if (response.getError().toString() != "") {
        System.out.println("Instance creation failed ! ! " + response.getError());
        return
      }
       System.out.println("Instance creation successful ! ! ");
```